readthedocs/support/views.py

```diff
@@ -0,0 +1,144 @@
+"""Support views."""
+
+import base64
+import hashlib
+import hmac
+import logging
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.db.models import Q
+from rest_framework.response import Response
+from rest_framework.status import (
+    HTTP_400_BAD_REQUEST,
+    HTTP_500_INTERNAL_SERVER_ERROR,
+)
+from rest_framework.views import APIView
+
+from readthedocs.core.utils.extend import SettingsOverrideObject
+from readthedocs.support.front import FrontClient
+
+log = logging.getLogger(__name__)
+
+
+class FrontWebhookBase(APIView):
+
+    """
+    Front's webhook handler.
+
+    Currently we only listen to inbound messages events.
+    Contact information is updated when a new message is received.
+
+    See https://dev.frontapp.com/docs/webhooks-1.
+    """
+
+    http_method_names = ['post']
+
+    def post(self, request):
+        if not self._is_payload_valid():
+            return Response(
+                {'detail': 'Invalid payload'},
+                status=HTTP_400_BAD_REQUEST,
+            )
+
+        event = request.data.get('type')
+        if event == 'inbound':
+            return self._update_contact_information(request.data)
+        return Response({'detail': f'Skipping {event} event'})
+
+    def _update_contact_information(self, data):
+        """
+        Update contact information using Front's API.
+
+        The webhook event give us the conversation_id,
+        we use that to
+        """
+        client = FrontClient(token=settings.FRONT_TOKEN)
+
+        # Retrieve the user from the email from the conversation.
+        conversation_id = data.get('conversation', {}).get('id')
+        try:
+            resp = client.get(f'/conversations/{conversation_id}').json()
+            email = resp.get('recipient', {}).get('handle')
+        except Exception:  # noqa
+            msg = f'Error while getting conversation {conversation_id}'
+            log.exception(msg)
+            return Response({'detail': msg}, status=HTTP_500_INTERNAL_SERVER_ERROR)
+
+        user = (
+            User.objects
+            .filter(Q(email=email) | Q(emailaddress__email=email))
+            .first()
+        )
+        if not user:
+            return Response({'detail': f'User with email {email} not found in our database'})
+
+        # Get current custom fields, and update them.
+        try:
+            resp = client.get(f'/contacts/alt:email:{email}').json()
+        except Exception:  # noqa
+            msg = f'Error while getting contact {email}'
+            log.exception(msg)
+            return Response({'detail': msg}, HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
Same here. 500 is not a good error IMO for external errors.

Also, the exception message could be improved by showing the URL tried: `url=...`